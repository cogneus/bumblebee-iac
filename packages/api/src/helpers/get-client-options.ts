import { ClientOptions } from 'bumblebee-common'
import { env } from 'process'
import { TemplatesParams } from '../types'

export const getClientOptions = ({ stage }: TemplatesParams): ClientOptions => {
  const {
    TABLE_TEMPLATES_LIVE,
    TABLE_TEMPLATES_STAGING,
    INDEX_GROUPS_LIVE,
    INDEX_GROUPS_STAGING,
    DB_LIMIT_DEFAULT,
    DB_LIMIT_MAX,
  } = env

  const tables = {
    live: TABLE_TEMPLATES_LIVE,
    staging: TABLE_TEMPLATES_STAGING,
  }

  const indices = {
    live: INDEX_GROUPS_LIVE,
    staging: INDEX_GROUPS_STAGING,
  }
  return {
    db: {
      maxAttempts: 1,
      region: '',
      requestTimeout: 3000,
      tables: {
        templates: {
          name: tables[stage || 'live'],
          itemLimits: {
            maxLimit: +DB_LIMIT_MAX,
            defaultLimit: +DB_LIMIT_DEFAULT,
          },
          indices: {
            groups: {
              name: indices[stage || 'live'],
            },
          },
        },
      },
    },
  }
}
