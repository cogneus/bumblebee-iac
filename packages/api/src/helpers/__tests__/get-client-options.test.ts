import { getClientOptions } from '../get-client-options'

describe('Check permissions tests', () => {
  beforeEach(() => {
    jest.clearAllMocks()
    process.env = {
      TABLE_TEMPLATES_LIVE: 'live-table',
      TABLE_TEMPLATES_STAGING: 'live-table',
      INDEX_GROUPS_LIVE: 'live-index',
      INDEX_GROUPS_STAGING: 'staging-index',
    } as any
  })

  test('should match client options for staging', () => {
    expect(getClientOptions({ stage: 'staging' })).toMatchSnapshot()
  })

  test('should match client options for live', () => {
    expect(getClientOptions({ stage: 'live' })).toMatchSnapshot()
    expect(getClientOptions({})).toMatchSnapshot()
  })
})
