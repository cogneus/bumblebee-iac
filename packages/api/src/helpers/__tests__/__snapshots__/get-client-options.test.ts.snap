// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check permissions tests should match client options for live 1`] = `
{
  "db": {
    "maxAttempts": 1,
    "region": "",
    "requestTimeout": 3000,
    "tables": {
      "templates": {
        "indices": {
          "groups": {
            "name": "live-index",
          },
        },
        "itemLimits": {
          "defaultLimit": NaN,
          "maxLimit": NaN,
        },
        "name": "live-table",
      },
    },
  },
}
`;

exports[`Check permissions tests should match client options for live 2`] = `
{
  "db": {
    "maxAttempts": 1,
    "region": "",
    "requestTimeout": 3000,
    "tables": {
      "templates": {
        "indices": {
          "groups": {
            "name": "live-index",
          },
        },
        "itemLimits": {
          "defaultLimit": NaN,
          "maxLimit": NaN,
        },
        "name": "live-table",
      },
    },
  },
}
`;

exports[`Check permissions tests should match client options for staging 1`] = `
{
  "db": {
    "maxAttempts": 1,
    "region": "",
    "requestTimeout": 3000,
    "tables": {
      "templates": {
        "indices": {
          "groups": {
            "name": "staging-index",
          },
        },
        "itemLimits": {
          "defaultLimit": NaN,
          "maxLimit": NaN,
        },
        "name": "live-table",
      },
    },
  },
}
`;
